ocf_datapipes/batch/merge_numpy_examples_to_batch.py

```diff
@@ -90,7 +90,7 @@ def stack_np_examples_into_batch(dict_list: Sequence[NumpyBatch]) -> NumpyBatch:
 
                 nwp_batch[nwp_source] = nwp_source_batch
 
-            batch[BatchKey.nwp] = nwp_batch
+            batch[BatchKey.nwp] = check_for_nans(nwp_batch)
 
```

Comment:
I might be wrong, but I was under the impression that we allow for nans currently to be present in batches, which then get filled with zeroes during training? @peterdudfield is this a gsp thing?

Reply:
Could we move this to when the NWP gets opened? And have an option to check it or not? 
I think that would make it safer and clearer whats going on. 

We could have a different issue that checks for nans in the batches, but we need to think how we turn that on and off .e.tc

Reply:
@glitch401 would you mind moving this to when the nwp is opened? with an option to do this or not.

Reply:
do you mean, when data element NWP is opened?

Reply:
Like below, in the load stage

Reply:
gotcha, will append changes

Reply:
i was hoping this would be removed, and it would be mvoed to below